build_tools/check_backticks.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+"""Test script to check for invalid use of single-backticks."""
+
+__author__ = ["geetu040"]
+
+import ast
+import os
+import re
+
+
+def find_py_files(folder_path):
+    """Find all Python files in a given folder path."""
+    py_files = []
+    for root, _, files in os.walk(folder_path):
+        for file in files:
+            if file.endswith(".py"):
+                py_files.append(os.path.join(root, file))
+    return py_files
+
+
+def extract_docstrings(filename):
+    """Extract docstrings from a Python file."""
+    with open(filename) as file:
+        tree = ast.parse(file.read(), filename=filename)
+
+    docstrings = {}
+    for node in ast.walk(tree):
+        if isinstance(node, (ast.FunctionDef, ast.ClassDef, ast.Module)):
+            if (
+                node.body
+                and isinstance(node.body[0], ast.Expr)
+                and isinstance(node.body[0].value, ast.Str)
+            ):
+                lineno = 0 if isinstance(node, ast.Module) else node.lineno
+                docstring = node.body[0].value.s
+                docstrings[lineno] = docstring
+    return docstrings
+
+
+def find_invalid_backtick_text(docstring):
+    """Find invalid backtick text in a docstring."""
+    # remove all the double-backticks
+    doc = docstring.replace("``", "")
```

Comment:
double-backticks are fine, problem occurs only with single-backticks. I am removing them so they don't interfere with the regex expressions. I am trying to keep the regex expressions as simple as possible and doing it in steps because they can become very complicated and hard to debug

*the updated code deals this differently and handles ```*